src/sagemaker/session.py

```diff
@@ -3284,17 +3284,22 @@ def _check_job_status(self, job, desc, status_key_name):
                 actual_status=status,
             )
 
-    def wait_for_endpoint(self, endpoint, poll=30):
+    def wait_for_endpoint(self, endpoint, poll=30, timeout_seconds=1800.0):
         """Wait for an Amazon SageMaker endpoint deployment to complete.
 
         Args:
             endpoint (str): Name of the ``Endpoint`` to wait for.
-            poll (int): Polling interval in seconds (default: 5).
+            poll (float): Polling interval in seconds (default: 30).
+            timeout_seconds (float): Timeout in seconds (default: 1800).
 
         Returns:
-            dict: Return value from the ``DescribeEndpoint`` API.
+            dict: Return value from the ``DescribeEndpoint`` API or None if timeout_seconds passed
         """
-        desc = _wait_until(lambda: _deploy_done(self.sagemaker_client, endpoint), poll)
+        desc = _wait_until(
+            lambda: _deploy_done(self.sagemaker_client, endpoint), poll, timeout_seconds
```

Comment:
can we also take this opportunity to rename deploy_done to deploy_status or something similar to keep it in same lines of naming as the other status check methods? Thanks!